src/main.rs

```diff
@@ -245,16 +105,23 @@ impl MainContext {
         let multi_progress = MultiProgress::new();
         let http_handling = args.http_handling.clone();
         let retry_policy = ExponentialBackoff::builder().build_with_max_retries(args.retries);
-        let client = ClientBuilder::new(
-            Client::builder()
-                .user_agent("aaoffline")
-                .https_only(http_handling == HttpHandling::Disallow)
-                .build()
-                .expect("client cannot be built"),
-        )
-        .with(RetryTransientMiddleware::new_with_policy(retry_policy))
-        .with_init(AaofflineMiddleware::from(&args))
-        .build();
+        let builder = Client::builder()
+            .user_agent("aaoffline")
+            .https_only(http_handling == args::HttpHandling::Disallow);
+        let builder = if args.connect_timeout > 0 {
+            builder.connect_timeout(Duration::from_secs(args.connect_timeout))
+        } else {
+            builder
+        };
+        let builder = if args.read_timeout > 0 {
+            builder.read_timeout(Duration::from_secs(args.read_timeout))
+        } else {
+            builder
+        };
```

Comment:
Can be written more concisely:

```suggestion
        let mut builder = Client::builder()
            .user_agent("aaoffline")
            .https_only(http_handling == args::HttpHandling::Disallow);
        if args.connect_timeout > 0 {
            builder = builder.connect_timeout(Duration::from_secs(args.connect_timeout));
        }
        if args.read_timeout > 0 {
            builder = builder.read_timeout(Duration::from_secs(args.read_timeout));
        }
```

Reply:
Oh right, that's a lot better.